pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java

```diff
@@ -132,6 +132,17 @@ protected Record<T> buildRecord(Consumer<T> consumer, Message<T> message) {
                 .message(message)
                 .schema(schema)
                 .topicName(message.getTopicName())
+                .customAckFunction(cumulative -> {
+                    try {
+                        if (cumulative) {
+                            consumer.acknowledgeCumulativeAsync(message);
+                        } else {
+                            consumer.acknowledgeAsync(message);
+                        }
+                    } finally {
+                        message.release();
```

Comment:
The ack is async, is it right to release here ?

Reply:
Thanks reply, i fixed it.